### ISSUE TITLE
install options have no effect?

### ISSUE THREAD

**Issue by haampie** [indygreg/python-zstandard#185]
I have the impression that none of this:

https://github.com/indygreg/python-zstandard/blob/80c3142f274621d11b1e3c401e17ee4b983ab1a5/setup.py#L56-L82

actually does anything.

I can pass `--install-option=--no-c-backend` but it is ignored.

**Comment by DjDeepDive** [comment]
Furthermore, checking for `sys.argv` directly is brittle, as PIP is deprecating the ability to pass arguments this way at all.  This should all be changed to `--config-settings` KEY=VALUE format if possible:

```
DEPRECATION: --build-option and --global-option are deprecated. pip 23.3 will enforce this behaviour change. A possible replacement is to use --config-settings. Discussion can be found at https://github.com/pypa/pip/issues/11859
```

**Comment by indygreg** [comment]
You need to use something like `--config-setting=--global-option=--no-c-backend` now. This does work with the latest uv/pip/setuptools. Our CI is now using this pattern. Docs have been updated. Closing issue.

**Comment by haampie** [comment]
For future readers, it requires pip 23.1 https://github.com/pypa/pip/commit/1268487cbab1b2ffab881b5961c0476e655b6c88 to pass multiple options:

```
pip install ... --config-setting=--global-option=--no-c-backend --config-setting=--global-option=--system-zstd
```

How the Python ecosystem landed on this design... I guess I should not be impressed.

**Comment by indygreg** [comment]
Yeah, I wrote a thorough blog post about this at https://gregoryszorc.com/blog/2023/10/30/my-user-experience-porting-off-setup.py/

2 years later and setuptools has yet to formalize how they'll handle config settings.

My latest realization is they pass through --build-option to the normal build wheel command (for backwards compatibility). But for editable wheels (PEP 660), they only pass through --global-option. Despite --build-option being semantically preferred. None of this is in the official documentation of course.

It makes no sense to me. But for the time being, --global-option does seem to work. If that stops working, hundreds of projects will be impacted. So I'm in good company.
